hack/generator/pkg/codegen/codegen_test.go

```diff
@@ -67,20 +67,29 @@ func makeTestLocalPackageReference(group string, version string) astmodel.LocalP
 	return astmodel.MakeLocalPackageReference(goModulePrefix, group, version)
 }
 
+func makeEmbeddedTestTypeDefinition() astmodel.TypeDefinition {
+	name := astmodel.MakeTypeName(makeTestLocalPackageReference("test", "v20200101"), "EmbeddedTestType")
+	t := astmodel.NewObjectType()
+	t.WithProperty(astmodel.NewPropertyDefinition("FancyProp", "fancyProp", astmodel.IntType))
```

Comment:
The `WithProperty()` method returns a new instance, which you're discarding. 
```suggestion
	t = t.WithProperty(astmodel.NewPropertyDefinition("FancyProp", "fancyProp", astmodel.IntType))
```

Reply:
oops!

Reply:
Fixed as part of another commit